packages/react-devtools-shared/src/devtools/views/Profiler/SnapshotSelector.js

```diff
@@ -35,7 +35,11 @@ export default function SnapshotSelector(_: Props) {
   const commitDurations: Array<number> = [];
   const commitTimes: Array<number> = [];
   commitData.forEach(commitDatum => {
-    commitDurations.push(commitDatum.duration);
+    commitDurations.push(
+      commitDatum.duration +
+        (commitDatum.effectDuration || 0) +
+        (commitDatum.passiveEffectDuration || 0),
+    );
```

Comment:
Where are these displayed? It's not obvious that one wants blocking (commit + layout effect) and non-blocking (passive effects) durations aggregated. I'm just guessing and say that these are responsible for color/height of the bars. Not sure if we always want to aggregate them. But adding UI/UX around separating blocking/non-blocking durations in these small bars might be difficult. For example, one could display each duration inside a bar but that might not be visible at all. And adding another option to just aggregate by blocking might be missed.

Reply:
Yes. This duration controls the height of the bars. I tried a UI treatment that separates the bars into 3 visible chunks and it was really weird looking so I ditched it. (I could share it here if you're curious to see it though.)

The duration tooltips and panel information are broken apart by each phase though so you can think about them separately if you want.

Reply:
Yeah, not sure if this will ever be a problem. In the end all of these durations will block anyway. Will see   if I prefer the old height calculation or the new one. But having effect duration available is already nice 👍

Reply:
Yeah essentially it doesn't work because there's not enough room to display the bars side by side (if there are a lot of commits) and displaying them stacked vertically often ends up making the commit phase bars too small to be visible (since their durations are so much smaller).

I _could_ try using a log/cbrt approach for durations here too but...

Reply:
Here's a screenshot from the test harness:
![Screen Shot 2021-03-16 at 2 57 27 PM](https://user-images.githubusercontent.com/29597/111364705-f54ab700-8667-11eb-8c65-da2f0108803f.png)

Reply:
Could also try to just give up on the color scale for these and show different sections (render vs layout vs passive) as different color bars:
![Screen Shot 2021-03-16 at 3 49 47 PM](https://user-images.githubusercontent.com/29597/111371009-41e5c080-866f-11eb-8b71-f36b46856d04.png)

Reply:
We could also use alpha/shading maybe
![Screen Shot 2021-03-16 at 4 19 07 PM](https://user-images.githubusercontent.com/29597/111374337-5b890700-8673-11eb-992c-08de09e7876d.png)

Reply:
I'm not sure any of these is a substantial improvement though. Maybe just more confusing at a glance.

Reply:
> I'm not sure any of these is a substantial improvement though. Maybe just more confusing at a glance.

It definitely looks cooler 😎 But I agree that this is probably more confusing.